BAZEL-bash.md

```diff
@@ -0,0 +1,162 @@
+string embedding a Bash script. Within that script, you need to use `$$` to
+produce a single `$` in the Bash script itself, as `$()` will be expanded by
+Bazel's own string interpolation. You can use the `location` and `locations`
+functions to get at file paths. You should note that the paths these returns is
+relative to the starting `$PWD` of the script, so if your Bash script `cd`s
+somwhere, you have to take that into account.
+
+Note that the script will start running from (virtually) the root of the
+project, regardless of where in the tree your package sits, in that if you
+include local files they will be under your package's path. Only the files you
+have explicitly specified as depencencies will be present.
+
+`srcs` files will generally be placed by Bazel at their current place in the
+workdir, e.g. if you specify `//:VERSION` you will get the `VERSION` file in
+your current directory when the script starts executing, but if you specify
+local targets they will be under the same path as your current `BUILD.bazel`
+file. Try to prefer using `location` entries when possible. I do not know how
+to do that for local files, though.
+
+You can specify local files (that are not Bazel rule results) using the `glob`
+function in the `srcs` attribute, e.g.
+```
+    srcs = glob([
+        "*.c"
+    ])
+```
+This function returns a Bazel array, which can be concatenated with another
+using `+` (typically a literal array to combine the list of local files with a
+list of Bazel targets).
+
+Note that tools and data dependencies will also be given as relative (to the
+script's starting directory) paths through the `location` function.
+
+## Conditionals
+
+Bazel supports simple conditionals in its rules, though I have not found a way
+to use them in string substitutions. As some rules have different names under
+Windows, this creates a bit of a problem. Aliases provide a way around that;
+for example:
+```
+alias(
+    name = "yarn",
+    actual = "@nodejs//:bin/yarn.cmd" if is_windows else "@nodejs//:bin/yarn",
+)
+```
+would let you use, in the `cmd` entry of a script:
+```
+        $$DIR/$(location :yarn)
+```
+to invoke the correct `yarn` executable regardless of platform.
+
+## Links and speed
+
+Bazel relies a lot on symbolic links to provide what should appear like a new,
+isolated environment to each build rule without having to pay the cost of
+copying files over each time. However, this means that
+1. Everything path-related is different on Windows, because symlinks there are
+   not well-supported, and
+1. While Bazel can claim some nice speed for creating the environment, every
+   read your rule does from a Bazel-provided file takes the hit of going
+   through the symlink indirection.
+That second point can be very costly when you have to read many small files
+(e.g. node dependencies).
+
+## `sh_test`: "runfiles"
+
+Life would be boring if all of the above applied to all invocations of Bash
```

Comment:
While I emphasize with your pain, I would prefer to limit those docs to the content that is actually helpful. This is already a fairly long document and adding more “prose” to it only makes it harder to find what you are looking for.